Set SONAME to libfec.so.0

While at it, link to libm to mute a dpkg-shlibdeps warnings

--- a/configure.in
+++ b/configure.in
@@ -83,7 +83,7 @@ darwin*)
 	REBIND=""
 	;;
 *)
-	SH_LIB=libfec.so
+	SH_LIB=libfec.so.0
 	REBIND=ldconfig
 	;;
 esac
--- a/makefile.in
+++ b/makefile.in
@@ -103,8 +103,8 @@ libfec.dylib: $(LIBS)
 	$(CC) -dynamiclib -install_name $@ -o $@ $^
 
 # for Linux et al
-libfec.so: $(LIBS)
-	$(CC) $(LDFLAGS) -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive -lc
+$(SHARED_LIB): $(LIBS)
+	$(CC) $(LDFLAGS) -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive -lc -lm
 
 dotprod.o: dotprod.c fec.h
 
